// @marcreichel/angular-carousel/esm2022/lib/carousel.module.mjs
import { CommonModule } from "@angular/common";
import { NgModule } from '@angular/core';
import { CarouselComponent } from './carousel.component';
import { CarouselSlideDirective } from "./slide.directive";
import * as i0 from "@angular/core";
export class CarouselModule {
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "16.1.7", ngImport: i0, type: CarouselModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule }); }
    static { this.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "14.0.0", version: "16.1.7", ngImport: i0, type: CarouselModule, declarations: [CarouselComponent,
            CarouselSlideDirective], imports: [CommonModule], exports: [CarouselComponent,
            CarouselSlideDirective] }); }
    static { this.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "16.1.7", ngImport: i0, type: CarouselModule, imports: [CommonModule] }); }
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "16.1.7", ngImport: i0, type: CarouselModule, decorators: [{
            type: NgModule,
            args: [{
                    declarations: [
                        CarouselComponent,
                        CarouselSlideDirective,
                    ],
                    imports: [
                        CommonModule,
                    ],
                    exports: [
                        CarouselComponent,
                        CarouselSlideDirective,
                    ]
                }]
        }] });
